### PR TITLE
Validate that custom ilm_policies are defined in the package

### PR DESCRIPTION
**Pull request by jsoriano**
Data streams can have custom ILM policies since https://github.com/elastic/package-spec/pull/90, but nothing is checking that these policies actually exist.

This new validation checks that the ILM policy is defined in the same data stream where it is used, what helps to have self-contained packages, and is coherent with the current use of this feature in existing packages.

Fixes https://github.com/elastic/package-spec/issues/92.

### CODE REVIEW

**Comment by jsoriano** [comment]
Migration docs added in https://github.com/elastic/elastic-package/pull/977.

File: test/packages/bad_custom_ilm_policy/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<!-- Use this template language as a starting point, replacing {placeholder text} with details about the integration. -->
+<!-- Find more detailed documentation guidelines in https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md -->
+
+# This package has a valid custom ILM policy.
```
Comment (mrodm) [inline test/packages/bad_custom_ilm_policy/docs/README.md:4]: Should the title be updated to mention that it has an invalid ILM Policy ? 

```suggestion
# This package has an invalid reference to a custom ILM policy.
```

If so, there are also some references in the following paragraphs

File: test/packages/custom_ilm_policy/changelog.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.0.1"
+  changes:
+    - description: Initial draft of the package
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/1 # FIXME Replace with the real PR link
```
Comment (mrodm) [inline test/packages/custom_ilm_policy/changelog.yml:6]: In the changelog of these test packages, should we update the links to be the PRs where they have been added ?

```suggestion
      link: https://github.com/elastic/integrations/pull/413
```

File: code/go/internal/validator/semantic/validate_ilmpolicypresent.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+import (
+	"fmt"
+	"io/fs"
+	"path"
+	"strings"
+
+	ve "github.com/elastic/package-spec/code/go/internal/errors"
+	"github.com/elastic/package-spec/code/go/internal/fspath"
+	"gopkg.in/yaml.v3"
```
Comment (mrodm) [inline code/go/internal/validator/semantic/validate_ilmpolicypresent.go:15]: nit: reorder imports
```suggestion
	"path"
	"strings"
	
	"gopkg.in/yaml.v3"

	ve "github.com/elastic/package-spec/code/go/internal/errors"
	"github.com/elastic/package-spec/code/go/internal/fspath"
```

**Review by mrodm — APPROVED** [review]
:+1: Added some non-blocking comments

**Review by mrodm — APPROVED** [review]

**Pull request merged**